zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -137,6 +156,23 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         );
 
         // Transfer value from caller to callee
+        let intrinsic_tx_value = cb.query_word();
+        cb.condition(is_tx_invalid.expr(), |cb| {
+            cb.require_equal(
+                "intrinsic_tx_value == 0",
+                intrinsic_tx_value.expr(),
+                0.expr(),
+            );
+        });
+
+        cb.condition(1.expr() - is_tx_invalid.expr(), |cb| {
+            cb.require_equal(
+                "intrinsic_tx_value == tx_value",
+                intrinsic_tx_value.expr(),
+                tx_value.expr(),
+            );
+        });
+
         let transfer_with_gas_fee = TransferWithGasFeeGadget::construct(
```

Comment:
Should transfer use intrinsic_tx_value??

Reply:
Good point. Updated.